src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -441,29 +446,32 @@ private void processTriggeredSegment(final RepairSegment segment, final JmxProxy
           segmentId,
           resultingSegment.getState());
 
-      if (RepairSegment.State.RUNNING == resultingSegment.getState()) {
-        LOG.info("Repair command {} on segment {} has been cancelled while running", this.repairNo, segmentId);
-        segmentFailed.set(true);
-        abort(resultingSegment, coordinator);
+      switch (resultingSegment.getState()) {
+        case RUNNING:
+          LOG.info("Repair command {} on segment {} has been cancelled while running", this.repairNo, segmentId);
+          segmentFailed.set(true);
+          abort(resultingSegment, coordinator);
+          break;
 
-      } else if (RepairSegment.State.DONE == resultingSegment.getState()) {
+        case DONE:
+          LOG.debug(
+              "Repair segment with id '{}' was repaired in {} seconds",
+              resultingSegment.getId(),
+              Seconds.secondsBetween(resultingSegment.getStartTime(), resultingSegment.getEndTime()).getSeconds());
 
-        LOG.debug(
-            "Repair segment with id '{}' was repaired in {} seconds",
-            resultingSegment.getId(),
-            Seconds.secondsBetween(resultingSegment.getStartTime(), resultingSegment.getEndTime()).getSeconds());
+          SEGMENT_RUNNERS.remove(resultingSegment.getId());
+          break;
 
-        SEGMENT_RUNNERS.remove(resultingSegment.getId());
-      } else {
-        // Something went wrong on the coordinator node and we never got the RUNNING notification
-        // or we are in an undetermined state.
-        // Let's just abort and reschedule the segment.
-        LOG.info(
-            "Repair command {} on segment {} never managed to start within timeout.",
-            this.repairNo,
-            segmentId);
-        segmentFailed.set(true);
-        abort(resultingSegment, coordinator);
+        default:
+          // Something went wrong on the coordinator node and we never got the RUNNING notification
+          // or we are in an undetermined state.
+          // Let's just abort and reschedule the segment.
+          LOG.info(
+              "Repair command {} on segment {} never managed to start within timeout.",
+              this.repairNo,
+              segmentId);
+          segmentFailed.set(true);
+          abort(resultingSegment, coordinator);
```

Comment:
this diff chunk isn't strictly part of the PR, it's just replacing the if-else blocks with a switch statement.